packages/migrate/src/__tests__/DbPull.test.ts

```diff
@@ -480,6 +481,947 @@ describe('postgresql - missing database', () => {
   })
 })
 
+describe('postgresql views re-introspection warnings', () => {
+  const connectionString = process.env.TEST_POSTGRES_URI_MIGRATE!.replace('tests-migrate', 'tests-migrate-db-pull')
+
+  function computeSetupParams(warningCode: EngineArgs.ViewWarningCodes) {
+    const setupParams: SetupParams = {
+      connectionString,
+      // Note: at this location there is a setup.sql file
+      // which will be executed a SQL file so the database is not empty
+      dirname: path.join(
+        __dirname,
+        '..',
+        '__tests__',
+        'fixtures',
+        're-introspection',
+        'postgresql',
+        `views-warning-${warningCode}`,
+      ),
+    }
+    return setupParams
+  }
+
+  describe('postgresql views 21', () => {
+    const warningCode = 21
+    const setupParams = computeSetupParams(warningCode)
+
+    beforeAll(async () => {
+      await tearDownPostgres(setupParams!).catch((e) => {
+        console.error(e)
+      })
+    })
+
+    beforeEach(async () => {
+      await setupPostgres(setupParams!).catch((e) => {
+        console.error(e)
+      })
+      // Back to original env vars
+      process.env = { ...originalEnv }
+      // Update env var because it's the one that is used in the schemas tested
+      process.env.TEST_POSTGRES_URI_MIGRATE = connectionString
+    })
+
+    afterEach(async () => {
```

Comment:
Feel free to ignore the suggestion if you disagree, but i think this setup is repeated for all of the tests? In that case, it could also be split into helpers:

```ts
function setupPostgressForWarining(warningCode: EngineArgs.ViewWarningCodes) {
  const setupParams = computeSetupParams(warningCode)
  
  beforeEach(async () => {
      await setupPostgres(setupParams!).catch((e) => {
        console.error(e)
      })
      // Back to original env vars
      process.env = { ...originalEnv }
      // Update env var because it's the one that is used in the schemas tested
      process.env.TEST_POSTGRES_URI_MIGRATE = connectionString
    })

    afterEach(async () => {
      // Back to original env vars
      process.env = { ...originalEnv }
      await tearDownPostgres(setupParams!).catch((e) => {
        console.error(e)
      })
    })
}
```

Reply:
Yep, absolutely agree!